engine/src/main/java/org/terasology/world/internal/WorldProviderCoreImpl.java

```diff
@@ -334,11 +334,12 @@ public int getExtraData(int index, int x, int y, int z) {
     public int setExtraData(int index, Vector3i worldPos, int value) {
         Vector3i chunkPos = ChunkMath.calcChunkPos(worldPos);
         CoreChunk chunk = chunkProvider.getChunk(chunkPos);
-        return setExtraData(index, chunk, worldPos,value);
+        return setExtraData(index, chunk, worldPos, value);
     }
 
     @Override
-    public int setExtraData(int index, CoreChunk chunk, Vector3i worldPos, int value) {
+    public int setExtraData(int index, CoreChunk chunk, Vector3i relPos, int value) {
+        Vector3i worldPos = chunk.chunkToWorldPosition(relPos);
         Vector3i chunkPos = ChunkMath.calcChunkPos(worldPos);
```

Comment:
What is chunkPos used for?

Reply:
It is not used after worldPos was added. Removed